src/runtime_src/tools/xclbinutil/SectionIPMetadata.cxx

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2018, 2022 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "SectionIPMetadata.h"
+
+#include "XclBinUtilities.h"
+#include <boost/format.hpp>
+#include <boost/functional/factory.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+namespace XUtil = XclBinUtilities;
+
+// Static Variables / Classes
+SectionIPMetadata::init SectionIPMetadata::initializer;
+
+SectionIPMetadata::init::init()
+{
+  auto sectionInfo = std::make_unique<SectionInfo>(IP_METADATA, "IP_METADATA", boost::factory<SectionIPMetadata*>());
+
+  sectionInfo->supportedAddFormats.push_back(FormatType::raw);
+
+  sectionInfo->supportedDumpFormats.push_back(FormatType::raw);
+
+  addSectionType(std::move(sectionInfo));
+}
+
+void
+SectionIPMetadata::marshalToJSON(char* _pDataSection,
```

Comment:
Going forward, we will no longer be using the underscore '_'prefixes for argument names.

Reply:
removed "_"prefixes